src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -468,8 +469,7 @@ def shape(self) -> tuple:
         """
         return (len(self),)
 
-    @is_lazy_check
-    def astype(self, dtype: Any, copy: bool = True) -> Index:
+    def astype(self, dtype: str | type | ExtensionDtype, copy: bool = True) -> Index:
```

Comment:
seems we removed the is_lazy_check for this API, is it now always lazy? feels we probably still need the is_lazy_check here, otherwise, what happens if the index is not lazily stored?

Reply:
Hmm that's a good point, the non-lazy functionality still exists. I added it back.